docs/dependency_support.md

```diff
@@ -6,18 +6,19 @@ usually at distro/user request.
 
 ## Linux kernel
 
-The linux kernel is bpftrace's biggest runtime dependency. Our stance on kernel
-support is that for **cross cutting** implementation details (ie. something
-every reasonably sophisticated user is expected to depend on), we'll support up
-to and including the oldest LTS kernel still in service.
+**Minimum kernel version: 5.15**
 
-For features that have clear boundaries (eg. new builtins or helpers), bpftrace
-is free to opportunistically depend on newer kernels as long as there is a
-reasonable runtime fallback strategy. An error message is a reasonable fallback
-strategy in this case.
+The linux kernel is bpftrace's biggest runtime dependency.
+While we would like to support the oldest LTS kernel version, we also want to make
+sure bpftrace is keeping up with the latest features. As a compromise bpftrace
+supports stable kernels and the 4 most recent LTS kernels.
```

Comment:
Optional: you could also just say something like "we support stable kernels and as many LTS kernels as we can, provided it doesn't cause significant complications (currently 4)".

You could also add explicitly: "users that require support for older kernels than that can often simply use older versions of bpftrace".

Reply:
I took the last suggestion but I like that we have some kind of concrete baseline (even if we have to adjust it in the future).